internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,168 @@ package github_connection
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
 	"github.com/google/go-github/v61/github"
 )
 
-type GithubConnectionParams struct {
-	RepositoryName    string
-	Token             string
-	RepoOwner         string
-	PullRequestNumber int
+type GithubConnection struct {
+	Client            *github.Client `json:"client"`
+	RepositoryName    string         `json:"repository_name"`
+	RepoOwner         string         `json:"repo_owner,omitempty"`
+	PullRequestNumber int            `json:"pull_request_number"`
+	EventName         string         `json:"event_name"`
+}
+
+type FileChange struct {
+	Filename    string
+	LineNumbers []int
+	Content     string
 }
```

Comment:
Code Review 2024-08-18 01:18:35 - New lines 6 to 17:

type GithubConnection struct {
	Client            *github.Client `json:"client"`
	RepositoryName    string         `json:"repository_name"`
	RepoOwner         string         `json:"repo_owner,omitempty"`
	PullRequestNumber int            `json:"pull_request_number"`
	EventName         string         `json:"event_name"`
}

type FileChange struct {
	Filename    string
	LineNumbers []int
	Content     string